tests/Custom/TimeoutTest.php

```diff
@@ -3,79 +3,82 @@
 namespace Mink\WebdriverClassicDriver\Tests\Custom;
 
 use Behat\Mink\Exception\DriverException;
-use Behat\Mink\Tests\Driver\TestCase;
-use Mink\WebdriverClassicDriver\WebdriverClassicDriver;
 
 class TimeoutTest extends TestCase
 {
-    /**
-     * @after
-     */
-    protected function resetSessions(): void
+    protected function tearDown(): void
     {
-        $session = $this->getSession();
-        $driver = $this->getSession()->getDriver();
-        assert($driver instanceof WebdriverClassicDriver);
-
-        // Stop the session instead of only resetting it, as timeouts are not reset (they are configuring the session itself)
-        if ($session->isStarted()) {
-            $session->stop();
-        }
-
-        // Reset the array of timeouts to avoid impacting other tests
-        $driver->setTimeouts([]);
+        $this->driver->setTimeouts([
+            'script' => 30000,
+            'page' => 300000,
+            'implicit' => 0,
+        ]);
```

Comment:
Why you've removed the `resetSessions` method which stopped the session (and reset timeouts) so that on next test in same test case you'll get a fresh driver?

What unstabilities in the test suite you're having, that forced you to rewrite this test case?

Reply:
I did that when I changed the base testcase, since it's not based on a mink session anymore.

I did it precisely because I believed that `resetSessions` wasn't working/behaving well enough:
1. it does not recreate the driver (e.g. driver state remains as is)
2. it cannot handle drivers created separately (not in a session)
3. I suspected that setting selenium timeout values stays permanent, even after the session is restarted
4. recreating the driver or manually calling `setTimeouts` (with an empty array) effectively doesn't do anything because of the way `applyTimeouts()` has been coded (which is why I put the defaults)

Anyway, a lot of that is mostly speculation. I'd have the other PR merged and then work on a solution here.

Reply:
I've approved that other PR. Haven't figured out, that most of code was coming it.

I've also understood the default timeout idea: since you're not creating a new driver instance in each test, then you need to change timeouts on the fly to make it work.

Reply:
Yes, but sadly it didn't fix the problem.

Reply:
By the way, since the weekend is almost over.. just wanted to say thanks for all the work and effort you've put in this weekend. @aik099

Reply:
> By the way, since the weekend is almost over.. just wanted to say thanks for all the work and effort you've put in this weekend. @aik099

@uuf6429 , my words exactly.